samples: Updating notes for central_smp_client readme (#7106)
nrfconnect/sdk-nrf: samples/bluetooth/central_smp_client/README.rst

The Central Simple Management Protocol (SMP) Client sample demonstrates how to use the :ref:`dfu_smp_readme` to connect to an SMP Server and send a simple echo command.
The response, which is received as CBOR-encoded data, is decoded and printed.

.. note::

>    This sample does *not* provide the means to program the device using DFU.
>    The sample is intended to show communication between SMP Client and SMP Server.

Reviewer: ```suggestion
   This sample does not provide the means to program a device using DFU.
   It demonstrates the communication between SMP Client and SMP Server.
```

The sample also requires a device running `mcumgr`_ with transport protocol over Bluetooth® Low Energy, for example, another development kit running the :ref:`smp_svr_sample`.

.. note::

>    This sample will *not* program the device using DFU.

Reviewer: ```suggestion
   This sample does not program the device using DFU.
```
